public/app/features/alerting/unified/components/rule-editor/ExpressionEditor.tsx

```diff
@@ -1,38 +1,53 @@
 import { noop } from 'lodash';
-import React, { FC, useCallback, useMemo, useState } from 'react';
-import { useAsync } from 'react-use';
+import React, { FC, useCallback, useMemo, useRef, useState } from 'react';
+import { useAsync, useObservable } from 'react-use';
 
-import { CoreApp, DataQuery } from '@grafana/data';
+import { CoreApp, DataQuery, DataSourceInstanceSettings } from '@grafana/data';
 import { getDataSourceSrv } from '@grafana/runtime';
+import { Button } from '@grafana/ui/src';
 import { LokiQuery } from 'app/plugins/datasource/loki/types';
 import { PromQuery } from 'app/plugins/datasource/prometheus/types';
 
+import { AlertingQueryRunner } from '../../state/AlertingQueryRunner';
+import { dataQueryToAlertQuery } from '../../utils/query';
+import { RuleViewerVisualization } from '../rule-viewer/RuleViewerVisualization';
+
 export interface ExpressionEditorProps {
   value?: string;
   onChange: (value: string) => void;
-  dataSourceName: string; // will be a prometheus or loki datasource
+  dsSettings: DataSourceInstanceSettings; // will be a prometheus or loki datasource
 }
 
-export const ExpressionEditor: FC<ExpressionEditorProps> = ({ value, onChange, dataSourceName }) => {
-  const { mapToValue, mapToQuery } = useQueryMappers(dataSourceName);
-  const [query, setQuery] = useState(mapToQuery({ refId: 'A', hide: false }, value));
+export const ExpressionEditor: FC<ExpressionEditorProps> = ({ value, onChange, dsSettings }) => {
+  const queryRunner = useRef(new AlertingQueryRunner());
+
+  const queryData = useObservable(queryRunner.current.get());
+
+  const { mapToValue, mapToQuery } = useQueryMappers(dsSettings.name);
+  const [dataQuery, setDataQuery] = useState(mapToQuery({ refId: 'A', hide: false }, value));
```

Comment:
What's the reason we have to use `{ refId: 'A', hide: false }`?

Reply:
This line no longer exists, but I'll try to explain. We needed an artificial `refId` value just to use the same function we use for Grafana-managed alerts (`mapToQuery`). In general, for cloud rules the `refId` is not used